testing/matrix/linter/run.go

```diff
@@ -52,6 +131,12 @@ func Run(tmpDir string, m utils.Module) error {
 
 	var values []string
 	var err error
+
+	err = changeSymlinks(m)
```

Comment:
For testing, it is better to use a proper way to setup and teardown a test suit. I previously proposed to use 
https://pkg.go.dev/github.com/stretchr/testify/suite#SetupTestSuite

It will help to restore symlinks in any case even if test failed to run.